### PR TITLE
GEM-121 | add decide endpoint

### PR DESCRIPTION
**Pull request by pwgacek**

### CODE REVIEW

File: src/integration/kotlin/pl/edu/agh/gem/integration/controler/ExternalPaymentControllerIT.kt [hunk 1]
```diff
@@ -164,4 +172,87 @@ class ExternalPaymentControllerIT(
...
+            // when
+            val response = service.decide(paymentDecisionRequest, createGemUser())
+
+            // then
+            response shouldHaveHttpStatus BAD_REQUEST
+            response shouldHaveValidationError expectedMessage
+        }
+    }
+
+    should("return forbidden if user is not a group member") {
```
Comment (lawos98) [inline src/integration/kotlin/pl/edu/agh/gem/integration/controler/ExternalPaymentControllerIT.kt:293]: we want to test that if we only have access to the recipent?

File: src/main/kotlin/pl/edu/agh/gem/external/dto/payment/PaymentDecisionRequest.kt [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import pl.edu.agh.gem.internal.model.payment.PaymentDecision
+import pl.edu.agh.gem.validation.ValidationMessage.GROUP_ID_NOT_BLANK
+import pl.edu.agh.gem.validation.ValidationMessage.MESSAGE_NULL_OR_NOT_BLANK
+import pl.edu.agh.gem.validation.ValidationMessage.PAYMENT_ID_NOT_BLANK
+
+data class PaymentDecisionRequest(
+    @field:NotBlank(message = PAYMENT_ID_NOT_BLANK)
+    val paymentId: String,
+    @field:NotBlank(message = GROUP_ID_NOT_BLANK)
+    val groupId: String,
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/external/dto/payment/PaymentDecisionRequest.kt:15]: same question like in pr before

File: src/main/kotlin/pl/edu/agh/gem/internal/service/PaymentService.kt [hunk 3]
```diff
@@ -56,7 +59,36 @@ class PaymentService(
 
     private fun getExchangeRate(baseCurrency: String, targetCurrency: String?, date: Instant) =
         targetCurrency?.let { currencyManagerClient.getExchangeRate(baseCurrency, targetCurrency, date) }?.value
+
+    fun decide(paymentDecision: PaymentDecision) {
+        val payment = paymentRepository.findByPaymentIdAndGroupId(paymentDecision.paymentId, paymentDecision.groupId)
+            ?: throw MissingPaymentException(paymentDecision.paymentId, paymentDecision.groupId)
+
+        if (payment.recipientId != paymentDecision.userId) {
```
Comment (lawos98) [inline src/main/kotlin/pl/edu/agh/gem/internal/service/PaymentService.kt]: can we just use validators?

File: src/test/kotlin/pl/edu/agh/gem/internal/service/PaymentServiceTest.kt [hunk 4]
```diff
@@ -125,6 +127,44 @@ class PaymentServiceTest : ShouldSpec({
...
+        whenever(paymentRepository.findByPaymentIdAndGroupId(PAYMENT_ID, GROUP_ID)).thenReturn(null)
+        val paymentDecision = createPaymentDecision()
+
+        // when & then
+        shouldThrowExactly<MissingPaymentException> { paymentService.decide(paymentDecision) }
+        verify(paymentRepository, times(1)).findByPaymentIdAndGroupId(PAYMENT_ID, GROUP_ID)
+        verify(paymentRepository, times(0)).save(anyVararg(Payment::class))
+    }
+
+    should("throw PaymentRecipientDecisionException when payment is not present") {
```
Comment (lawos98) [inline src/test/kotlin/pl/edu/agh/gem/internal/service/PaymentServiceTest.kt]: fix for test name

File: src/integration/kotlin/pl/edu/agh/gem/integration/controler/ExternalPaymentControllerIT.kt (hunk 1 above)
Comment (pwgacek) [reply src/integration/kotlin/pl/edu/agh/gem/integration/controler/ExternalPaymentControllerIT.kt:293]: as told on daily checking if user belongs to the group stays so this test is needed

**Review by lawos98 — APPROVED** [review]

**Pull request merged**